packages/contracts/test/framework/plugin/plugin-repo.ts

```diff
@@ -61,212 +46,442 @@ describe.skip('PluginRepo', function () {
+      const randomContract = await (
+        await ethers.getContractFactory('TestPlugin')
+      ).deploy();
+      await expect(
+        pluginRepo.createVersion(
+          1,
+          randomContract.address,
+          emptyBytes,
+          emptyBytes
+        )
+      ).to.be.revertedWithCustomError(
+        pluginRepo,
+        'InvalidPluginSetupInterface'
+      );
+    });
 
-  // valid version as being a correct bump from 0.0.0
-  it('cannot create invalid first version', async function () {
-    await expect(
-      pluginRepo.createVersion([1, 1, 0], pluginSetupMock.address, emptyBytes)
-    )
-      .to.be.revertedWithCustomError(pluginRepo, 'BumpInvalid')
-      .withArgs([0, 0, 0], [1, 1, 0]);
-  });
+    it('fails if the release number is 0', async () => {
+      await expect(
+        pluginRepo.createVersion(
+          0,
+          pluginSetupMock.address,
+          emptyBytes,
+          emptyBytes
+        )
+      ).to.be.revertedWithCustomError(pluginRepo, 'ReleaseZeroNotAllowed');
+    });
 
-  it('cannot create version with unsupported interface contract', async function () {
-    // Use the `DAO` contract for testing purposes here, because the interface differs from `PluginSetup`.
-    const DAO = await ethers.getContractFactory('DAO');
-    let contractNotBeingAPluginSetup = await DAO.deploy();
-    await expect(
-      pluginRepo.createVersion(
-        [1, 0, 0],
-        contractNotBeingAPluginSetup.address,
-        emptyBytes
-      )
-    )
-      .to.be.revertedWithCustomError(pluginRepo, 'InvalidPluginSetupInterface')
-      .withArgs(contractNotBeingAPluginSetup.address);
-  });
+    it('fails if the release is incremented by more than 1', async () => {
+      await pluginRepo.createVersion(
+        1,
+        pluginSetupMock.address,
+        BUILD_METADATA,
+        RELEASE_METADATA
+      );
 
-  it('cannot create version with random address', async function () {
-    const randomAddress = await signers[8].getAddress();
+      await expect(
+        pluginRepo.createVersion(
+          3,
+          pluginSetupMock.address,
+          BUILD_METADATA,
+          RELEASE_METADATA
+        )
+      ).to.be.revertedWithCustomError(pluginRepo, 'InvalidReleaseIncrement');
+    });
 
-    await expect(
-      pluginRepo.createVersion([1, 0, 0], randomAddress, emptyBytes)
-    ).to.be.revertedWith('Address: call to non-contract');
-  });
+    it('fails for the first release, if `releaseMetadata` is empty', async () => {
+      await expect(
+        pluginRepo.createVersion(
+          1,
+          pluginSetupMock.address,
+          BUILD_METADATA,
+          '0x'
+        )
+      ).to.be.revertedWithCustomError(pluginRepo, 'EmptyReleaseMetadata');
+    });
 
-  context('creating initial version', async function () {
-    let initialPluginSetup: any;
-    const initialContent = '0x12';
+    it('fails if the same plugin setup exists in another release', async () => {
```

Comment:
Shouldn't we make transparent / state more explicitly that it only checks
`Version storage version = versions[latestTagHashForPluginSetup[_pluginSetup]];`, also on the contreact?

Otherwise, people will think it does a complete check